src/sysext/sysext.c

```diff
@@ -169,6 +259,16 @@ static int unmerge_hierarchy(const char *p) {
 
 static int unmerge(void) {
         int r, ret = 0;
+        bool need_to_reload = false;
+
+        if (!arg_no_reload) {
+                int reload;
+
+                reload = need_reload();
+                if (reload < 0)
+                        return log_debug_errno(reload, "Unable to verify if the service manager needs a reload: %m");
```

Comment:
need_reload() already logged about all its errors at a higher log level, no need to repeat this here at debug log level. just do `return reload;` here in case of error.